buildkite/resource_cluster_queue.go

```diff
@@ -0,0 +1,245 @@
+				Optional:            true,
+				MarkdownDescription: "A description for the Cluster Queue. ",
+			},
+		},
+	}
+}
+
+func (cq *ClusterQueueResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan, state ClusterQueueResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	log.Printf("Creating cluster queue with key %s into cluster %s ...", plan.Key.ValueString(), plan.ClusterId.ValueString())
+	apiResponse, err := createClusterQueue(
+		cq.client.genqlient,
+		cq.client.organizationId,
+		plan.ClusterId.ValueString(),
+		plan.Key.ValueString(),
+		plan.Description.ValueStringPointer(),
+	)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to create Cluster Queue",
+			fmt.Sprintf("Unable to create Cluster Queue: %s", err.Error()),
+		)
+		return
+	}
+
+	state.Id = types.StringValue(apiResponse.ClusterQueueCreate.ClusterQueue.Id)
+	state.Uuid = types.StringValue(apiResponse.ClusterQueueCreate.ClusterQueue.Uuid)
+	state.ClusterId = plan.ClusterId
+	state.ClusterUuid = types.StringValue(apiResponse.ClusterQueueCreate.ClusterQueue.Cluster.Uuid)
+	state.Key = types.StringValue(apiResponse.ClusterQueueCreate.ClusterQueue.Key)
+	state.Description = types.StringPointerValue(apiResponse.ClusterQueueCreate.ClusterQueue.Description)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (cq *ClusterQueueResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state ClusterQueueResourceModel
+	var queueFound bool = false
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	log.Printf("Getting cluster queues for cluster %s ...", state.ClusterUuid.ValueString())
+	queues, err := getClusterQueues(cq.client.genqlient, cq.client.organization, state.ClusterUuid.ValueString())
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to read Cluster Queues",
+			fmt.Sprintf("Unable to read Cluster Queues: %s", err.Error()),
+		)
+		return
+	}
+
+	// Find the cluster queue from the returned queues to update state
+	for _, edge := range queues.Organization.Cluster.Queues.Edges {
+		if edge.Node.Id == state.Id.ValueString() {
+			log.Printf("Found cluster queue with ID %s in cluster %s", edge.Node.Id, state.ClusterUuid.ValueString())
+			// Update ClusterQueueResourceModel with Node values and append
+			updateClusterQueueResource(edge.Node, &state)
+			resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+			return
+		}
+	}
+
+	// If not returned by this point, the cluster queue could not be found
+	// This is a tradeoff of the current getClusterQueues Genqlient query (searches for 50 queues via the cluster UUID in state)
+	if !queueFound {
+		resp.Diagnostics.AddError(
+			"Unable to find Cluster Queue",
+			fmt.Sprintf("Unable to find Cluster Queue: %s", err.Error()),
+		)
+		return
+	}
```

Comment:
This doesn't require a conditional for 2 reasons; 1 is that queueFound is never set to `true`, so checking that its value has changed is redundant.

The second is that on a number of occasions (line 127, 138 and 148) the `Read` function is exited, so the program cannot make it to this point and not be in a position to produce an error. Thus, the below should behave the same way.

```go
resp.Diagnostics.AddError(
    "Unable to find Cluster Queue",
	fmt.Sprintf("Unable to find Cluster Queue: %s", err.Error()),
)
return
```

Reply:
Example: https://go.dev/play/p/XYAsrs7q8u0

Reply:
Thinking it would be the case - amended 👍